### ISSUE TITLE
Cluster - Failed to get master URL

### ISSUE THREAD

**Issue by p30arena** [arangodb/arangodb#5373]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.3.8

Mode:
- [x] Cluster

Storage-Engine:
- [x] rocksdb

On this operating system:
- [ ] Linux 
  - [ ] Centos 7 .rpm

first I issued the arangodb command, the cluster was up and runnnig,
then we encountered a power failure,
after restart I saw that arangodb was listening to port 8529,
so I killed the process and issued the arangodb command again,
this is the output I'm receiving:

```
2018/05/16 09:54:13 Starting arangodb version 0.11.2, build 4d6e075
2018/05/16 09:54:13 Relaunching service with id 'ed10f41b' on :8528...
2018/05/16 09:54:13 ArangoDB Starter listening on 0.0.0.0:8528 (:8528)
2018/05/16 09:54:13 Looking for a running instance of agent on port 8531
2018/05/16 09:54:13 Starting agent on port 8531
2018/05/16 09:54:14 Looking for a running instance of dbserver on port 8530
2018/05/16 09:54:14 Starting dbserver on port 8530
2018/05/16 09:54:15 Looking for a running instance of coordinator on port 8529
2018/05/16 09:54:15 Starting coordinator on port 8529
2018/05/16 09:54:19 agent up and running (version 3.3.8).
2018/05/16 09:55:23 Failed to get master URL, retrying in 5sec (context deadline exceeded)
2018/05/16 09:55:38 Failed to get master URL, retrying in 5sec (context deadline exceeded)
2018/05/16 09:55:53 Failed to get master URL, retrying in 5sec (context deadline exceeded)
2018/05/16 09:56:08 Failed to get master URL, retrying in 5sec (context deadline exceeded)
```

**Comment by p30arena** [comment]
some new logs today for coordinator:


```
2018-05-20T00:10:07Z [15428] INFO ArangoDB 3.3.8 [linux] 64bit, using jemalloc, build tags/v3.3.8-0-g50ef880, VPack 0.1.30, RocksDB 5.6.0, ICU 58.1, V8 5.7.492.77, OpenSSL 1.0.2k-fips  26 Jan 2017
2018-05-20T00:10:08Z [15428] INFO {authentication} Jwt secret not specified, generating...
2018-05-20T00:10:08Z [15428] INFO detected operating system: Linux version 3.10.0-514.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-11) (GCC) ) #1 SMP Tue Nov 22 16:42:41 UTC 2016
2018-05-20T00:10:08Z [15428] WARNING {memory} environment variable GLIBCXX_FORCE_NEW' is not set. it is recommended to set it to some value to avoid unnecessary memory pooling in glibc++
2018-05-20T00:10:08Z [15428] WARNING {memory} execute 'export GLIBCXX_FORCE_NEW=1'
2018-05-20T00:10:08Z [15428] WARNING {memory} /sys/kernel/mm/transparent_hugepage/enabled is set to 'always'. It is recommended to set it to a value of 'never' or 'madvise'
2018-05-20T00:10:08Z [15428] WARNING {memory} /sys/kernel/mm/transparent_hugepage/defrag is set to 'always'. It is recommended to set it to a value of 'never' or 'madvise'
2018-05-20T00:10:08Z [15428] WARNING {memory} execute 'sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/enabled"'
2018-05-20T00:10:08Z [15428] WARNING {memory} execute 'sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/defrag"'
2018-05-20T00:10:08Z [15428] INFO using storage engine mmfiles
2018-05-20T00:10:08Z [15428] INFO {cluster} Starting up with role COORDINATOR
2018-05-20T00:10:20Z [15428] INFO {agencycomm} Flaky agency communication to http+tcp://192.168.136.132:8531. Unsuccessful consecutive tries: 21 (11.33s). Network checks advised.
```

**Comment by p30arena** [comment]
agent

```
2018-05-20T00:15:14Z [15472] INFO ArangoDB 3.3.8 [linux] 64bit, using jemalloc, build tags/v3.3.8-0-g50ef880, VPack 0.1.30, RocksDB 5.6.0, ICU 58.1, V8 5.7.492.77, OpenSSL 1.0.2k-fips  26 Jan 2017
2018-05-20T00:15:14Z [15472] INFO {authentication} Jwt secret not specified, generating...
2018-05-20T00:15:14Z [15472] INFO detected operating system: Linux version 3.10.0-514.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-11) (GCC) ) #1 SMP Tue Nov 22 16:42:41 UTC 2016
2018-05-20T00:15:14Z [15472] WARNING {memory} environment variable GLIBCXX_FORCE_NEW' is not set. it is recommended to set it to some value to avoid unnecessary memory pooling in glibc++
2018-05-20T00:15:14Z [15472] WARNING {memory} execute 'export GLIBCXX_FORCE_NEW=1'
2018-05-20T00:15:14Z [15472] WARNING {memory} /sys/kernel/mm/transparent_hugepage/enabled is set to 'always'. It is recommended to set it to a value of 'never' or 'madvise'
2018-05-20T00:15:14Z [15472] WARNING {memory} /sys/kernel/mm/transparent_hugepage/defrag is set to 'always'. It is recommended to set it to a value of 'never' or 'madvise'
2018-05-20T00:15:14Z [15472] WARNING {memory} execute 'sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/enabled"'
2018-05-20T00:15:14Z [15472] WARNING {memory} execute 'sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/defrag"'
2018-05-20T00:15:14Z [15472] INFO using storage engine mmfiles
2018-05-20T00:15:14Z [15472] INFO {cluster} Starting up with role AGENT
2018-05-20T00:15:14Z [15472] INFO {syscall} file-descriptors (nofiles) hard limit is 8192, soft limit is 8192
2018-05-20T00:15:14Z [15472] INFO {authentication} Authentication is turned off, authentication for unix sockets is turned on
2018-05-20T00:15:21Z [15472] INFO using endpoint 'http+tcp://[::]:8531' for non-encrypted requests
2018-05-20T00:15:26Z [15472] INFO ArangoDB (version 3.3.8 [linux]) is ready for business. Have fun!
2018-05-20T00:15:26Z [15472] INFO {agency} Restarting agent from persistence ...
2018-05-20T00:15:26Z [15472] INFO {agency} Found majority of agents in agreement over active pool. Finishing startup sequence.
2018-05-20T00:15:26Z [15472] INFO {agency} Activating agent with pool { AGNT-c7e9ca21-3bd8-44ec-b4e0-81f6c7afd589: tcp://192.168.136.132:8531, AGNT-df1e6194-80bf-4a42-8759-5b0c56872f3b: tcp://192.168.136.131:8531, AGNT-4924918e-45da-46c2-b69a-dc7e7e1de711: tcp://192.168.136.130:8531 }
```

**Comment by p30arena** [comment]
Its good to say that I've tested the scenario twice on fresh directories, on single and multiple servers.
I did also disable the arangodb3 service before turning off the servers, so when the server starts again the default arangodb3 service is not running on 8529.
`systemctl disable arangodb3`

**Comment by p30arena** [comment]
@dothebart do you suggest manually configuring the cluster?
its been about a week that I've been waiting for an answer

**Comment by OmarAyo** [comment]
Hi @p30arena 

Sorry for the delayed response

In order to proceed with our analysis, could you please provide us the following:
- exact command you are using to deploy the ArangoDB Cluster from the Starter
- Storage-engine: you did not specify it so I guess that it could be mmfiles, if so we suggest you to move to RocksDB 
- agency dump (we can help you to get it if you need)

Best,

Omar

**Comment by OmarAyo** [comment]
Hi @p30arena

In addition to what I said before:

- In order to have an agency dump you can send the following request:
  `curl -L http://IP:PORT/_api/agency/read -L -d '[[ "/" ]]' --dump -`
  Please make sure to use the IP and PORT of one agent
  
In case you are using authentication you should convert your "JWT-secret" into a token, this can be done in the following way:

- Let's say the JWT-secret is _geheim_
- Run the following command:  `jwtgen -a HS256 -s geheim -c server_id=hans -c iss=arangodb`

And your output will be: `eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE1MjIzMjY2NzksInNlcnZlcl9pZCI6ImhhbnMiLCJpc3MiOiJhcmFuZ29kYiJ9.6VwdMM_tEfMWgBzdjJ567paHhxjDXIKC1wINwNbQciQ`

Note that the server_id and iss values are essentially arbitrary, the crucial thing is the HS256 and the JWT secret geheim. This needs a node.js program called jwtgen which one can install on a system with node.js as follows:

`npm install -g jwtgen`

After you get your token simply send the following request:

`curl -L http://IP:PORT/_api/agency/read -d '[["/"]]' -H "Authentication: bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE1MjIzMjY2NzksInNlcnZlcl9pZCI6ImhhbnMiLCJpc3MiOiJhcmFuZ29kYiJ9.6VwdMM_tEfMWgBzdjJ567paHhxjDXIKC1wINwNbQciQ"`

(In the command above please use the token that you have generated eariler)

- In case you want to switch to RocksDB storage engine (and we suggest you to do that) you can add the following option to the Starter when you start your ArangoDB Cluster: `--server.storage-engine=rocksdb `. Please note that this option should be used on fresh Cluster directories

Best,

**Comment by p30arena** [comment]
`HTTP/1.1 503 Service Unavailable
X-Content-Type-Options: nosniff
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 23

{"message":"No leader"}`

**Comment by p30arena** [comment]
and retrying with another server running

```
HTTP/1.1 307 Temporary Redirect
X-Content-Type-Options: nosniff
Location: http://192.168.136.131:8531/_api/agency/read
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 0

HTTP/1.1 200 OK
X-Content-Type-Options: nosniff
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 18639

[{"arango":{"Current":{"Singles":{},"Databases":{"_system":{"PRMR-d7b9e6c2-e2ac-4c4c-bcb8-59eddaf182fd":{"name":"_system","error":false,"id":"1","errorNum":0,"errorMessage":""},"PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a":{"error":false,"name":"_system","id":"1","errorMessage":"","errorNum":0},"PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1":{"error":false,"name":"_system","id":"1","errorMessage":"","errorNum":0}}},"ShardsCopied":{},"DBServers":{"PRMR-d7b9e6c2-e2ac-4c4c-bcb8-59eddaf182fd":"none","PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1":"none","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a":"none"},"Collections":{"_system":{"3010026":{"s3010027":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010021":{"s3010022":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["time"],"id":"3010023","sparse":false,"type":"skiplist","unique":false}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010015":{"s3010016":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["time"],"id":"3010017","sparse":false,"type":"skiplist","unique":false}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010028":{"s3010029":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["queue","status","delayUntil"],"id":"3010030","sparse":false,"type":"skiplist","unique":true},{"deduplicate":true,"fields":["status","queue","delayUntil"],"id":"3010031","sparse":false,"type":"skiplist","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010006":{"s3010007":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010024":{"s3010025":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010032":{"s3010033":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["mount"],"id":"3010034","sparse":false,"type":"hash","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010035":{"s3010036":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"5010002":{"s5010003":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1"],"error":false,"errorNum":0,"errorMessage":""}},"3010008":{"s3010009":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010018":{"s3010019":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["time"],"id":"3010020","sparse":false,"type":"skiplist","unique":false}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010003":{"s3010004":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["user"],"id":"3010005","sparse":true,"type":"hash","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010001":{"s3010002":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}},"3010013":{"s3010014":{"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"error":false,"servers":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"],"errorNum":0,"errorMessage":""}}}},"Foxxmaster":"CRDN-d00e0350-8108-456f-802c-53c3a8b54174","FoxxmasterQueueupdate":false,"Coordinators":{"CRDN-0ea2f097-d05e-4b43-968a-810c87146354":"none","CRDN-83ec9ecb-a66e-45df-8ce6-aa677fd2d92e":"none","CRDN-d00e0350-8108-456f-802c-53c3a8b54174":"none"},"Version":69,"AsyncReplication":{},"NewServers":{},"Lock":"UNLOCKED","ServersRegistered":{"PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a":{"endpoint":"tcp://192.168.136.130:8530","host":"e9f0b52f06194937928cbb1f3ed719aa"},"PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1":{"endpoint":"tcp://192.168.136.131:8530","host":"e9f0b52f06194937928cbb1f3ed719aa"},"Version":1,"CRDN-0ea2f097-d05e-4b43-968a-810c87146354":{"endpoint":"tcp://192.168.136.130:8529","host":"e9f0b52f06194937928cbb1f3ed719aa"},"CRDN-d00e0350-8108-456f-802c-53c3a8b54174":{"endpoint":"tcp://192.168.136.131:8529","host":"e9f0b52f06194937928cbb1f3ed719aa"},"CRDN-83ec9ecb-a66e-45df-8ce6-aa677fd2d92e":{"endpoint":"tcp://192.168.136.132:8529","host":"e9f0b52f06194937928cbb1f3ed719aa"},"PRMR-d7b9e6c2-e2ac-4c4c-bcb8-59eddaf182fd":{"endpoint":"tcp://192.168.136.132:8530","host":"e9f0b52f06194937928cbb1f3ed719aa"}}},"Cluster":"9c702d93-7c86-4460-a88c-79439e91d4e9","SystemCollectionsCreated":true,"InitDone":true,"Target":{"NumberOfDBServers":null,"Pending":{},"Lock":"UNLOCKED","Finished":{"9030013":{"timeCreated":"2018-05-30T10:18:08Z","server":"PRMR-d7b9e6c2-e2ac-4c4c-bcb8-59eddaf182fd","jobId":"9030013","timeStarted":"2018-05-30T10:18:09Z","timeFinished":"2018-05-30T10:18:10Z","type":"failedServer","creator":"supervision"}},"LatestDBServerId":3,"MapLocalToID":{},"Failed":{},"FailedServers":{},"MapUniqueToShortID":{"CRDN-d00e0350-8108-456f-802c-53c3a8b54174":{"ShortName":"Coordinator0003","TransactionID":3},"PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a":{"ShortName":"DBServer0003","TransactionID":3},"PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1":{"ShortName":"DBServer0002","TransactionID":2},"CRDN-0ea2f097-d05e-4b43-968a-810c87146354":{"ShortName":"Coordinator0001","TransactionID":1},"PRMR-d7b9e6c2-e2ac-4c4c-bcb8-59eddaf182fd":{"ShortName":"DBServer0001","TransactionID":1},"CRDN-83ec9ecb-a66e-45df-8ce6-aa677fd2d92e":{"ShortName":"Coordinator0002","TransactionID":2}},"CleanedServers":[],"ToDo":{},"NumberOfCoordinators":null,"Version":1,"LatestCoordinatorId":3},"Sync":{"HeartbeatIntervalMs":1000,"UserVersion":1,"ServerStates":{},"Commands":{},"Problems":{},"LatestID":12040001},"Plan":{"DBServers":{"PRMR-d7b9e6c2-e2ac-4c4c-bcb8-59eddaf182fd":"none","PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1":"none","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a":"none"},"Collections":{"_system":{"3010026":{"journalSize":1048576,"isSmart":false,"numberOfShards":1,"isSystem":true,"isVolatile":false,"id":"3010026","keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"indexBuckets":8,"deleted":false,"waitForSync":false,"statusString":"loaded","type":2,"shards":{"s3010027":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","doCompact":true,"name":"_queues","shardKeys":["_key"],"distributeShardsLike":"3010001"},"3010021":{"deleted":false,"waitForSync":false,"distributeShardsLike":"3010001","shardKeys":["_key"],"doCompact":true,"id":"3010021","isVolatile":false,"keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexBuckets":8,"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["time"],"id":"3010023","sparse":false,"type":"skiplist","unique":false}],"statusString":"loaded","type":2,"isSystem":true,"shards":{"s3010022":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","name":"_statistics15","isSmart":false,"journalSize":8388608,"numberOfShards":1},"3010015":{"deleted":false,"waitForSync":false,"distributeShardsLike":"3010001","shardKeys":["_key"],"doCompact":true,"id":"3010015","isVolatile":false,"keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexBuckets":8,"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["time"],"id":"3010017","sparse":false,"type":"skiplist","unique":false}],"statusString":"loaded","type":2,"isSystem":true,"shards":{"s3010016":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","name":"_statisticsRaw","isSmart":false,"journalSize":8388608,"numberOfShards":1},"3010028":{"deleted":false,"waitForSync":false,"distributeShardsLike":"3010001","shardKeys":["_key"],"doCompact":true,"id":"3010028","isVolatile":false,"keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexBuckets":8,"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["queue","status","delayUntil"],"id":"3010030","sparse":false,"type":"skiplist","unique":true},{"deduplicate":true,"fields":["status","queue","delayUntil"],"id":"3010031","sparse":false,"type":"skiplist","unique":true}],"statusString":"loaded","type":2,"isSystem":true,"shards":{"s3010029":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","name":"_jobs","isSmart":false,"journalSize":2097152,"numberOfShards":1},"3010006":{"journalSize":1048576,"isSmart":false,"numberOfShards":1,"isSystem":true,"isVolatile":false,"id":"3010006","keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"indexBuckets":8,"deleted":false,"waitForSync":false,"statusString":"loaded","type":2,"shards":{"s3010007":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","doCompact":true,"name":"_modules","shardKeys":["_key"],"distributeShardsLike":"3010001"},"3010024":{"journalSize":1048576,"isSmart":false,"numberOfShards":1,"isSystem":true,"isVolatile":false,"id":"3010024","keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"indexBuckets":8,"deleted":false,"waitForSync":false,"statusString":"loaded","type":2,"shards":{"s3010025":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","doCompact":true,"name":"_frontend","shardKeys":["_key"],"distributeShardsLike":"3010001"},"3010032":{"deleted":false,"waitForSync":false,"distributeShardsLike":"3010001","shardKeys":["_key"],"doCompact":true,"id":"3010032","isVolatile":false,"keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexBuckets":8,"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["mount"],"id":"3010034","sparse":false,"type":"hash","unique":true}],"statusString":"loaded","type":2,"isSystem":true,"shards":{"s3010033":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","name":"_apps","isSmart":false,"journalSize":2097152,"numberOfShards":1},"3010035":{"journalSize":2097152,"isSmart":false,"numberOfShards":1,"isSystem":true,"isVolatile":false,"id":"3010035","keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"indexBuckets":8,"deleted":false,"waitForSync":false,"statusString":"loaded","type":2,"shards":{"s3010036":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","doCompact":true,"name":"_appbundles","shardKeys":["_key"],"distributeShardsLike":"3010001"},"5010002":{"keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"isVolatile":false,"id":"5010002","journalSize":33554432,"isSmart":false,"numberOfShards":1,"isSystem":true,"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"deleted":false,"waitForSync":false,"shardKeys":["_key"],"name":"_fishbowl","statusString":"loaded","type":2,"status":3,"shards":{"s5010003":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1"]},"replicationFactor":1,"indexBuckets":8,"path":"","doCompact":true},"3010008":{"journalSize":4194304,"isSmart":false,"numberOfShards":1,"isSystem":true,"isVolatile":false,"id":"3010008","keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"indexBuckets":8,"deleted":false,"waitForSync":false,"statusString":"loaded","type":2,"shards":{"s3010009":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","doCompact":true,"name":"_routing","shardKeys":["_key"],"distributeShardsLike":"3010001"},"3010018":{"deleted":false,"waitForSync":false,"distributeShardsLike":"3010001","shardKeys":["_key"],"doCompact":true,"id":"3010018","isVolatile":false,"keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexBuckets":8,"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["time"],"id":"3010020","sparse":false,"type":"skiplist","unique":false}],"statusString":"loaded","type":2,"isSystem":true,"shards":{"s3010019":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","name":"_statistics","isSmart":false,"journalSize":8388608,"numberOfShards":1},"3010003":{"deleted":false,"waitForSync":false,"distributeShardsLike":"3010001","shardKeys":["user"],"doCompact":true,"id":"3010003","isVolatile":false,"keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexBuckets":8,"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true},{"deduplicate":true,"fields":["user"],"id":"3010005","sparse":true,"type":"hash","unique":true}],"statusString":"loaded","type":2,"isSystem":true,"shards":{"s3010004":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","name":"_users","isSmart":false,"journalSize":4194304,"numberOfShards":1},"3010001":{"keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"isVolatile":false,"id":"3010001","journalSize":1048576,"isSmart":false,"numberOfShards":1,"isSystem":true,"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"deleted":false,"waitForSync":false,"shardKeys":["_key"],"name":"_graphs","statusString":"loaded","type":2,"status":3,"shards":{"s3010002":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"replicationFactor":2,"indexBuckets":8,"path":"","doCompact":true},"3010013":{"journalSize":1048576,"isSmart":false,"numberOfShards":1,"isSystem":true,"isVolatile":false,"id":"3010013","keyOptions":{"allowUserKeys":true,"lastValue":0,"type":"traditional"},"indexes":[{"fields":["_key"],"id":"0","sparse":false,"type":"primary","unique":true}],"indexBuckets":8,"deleted":false,"waitForSync":false,"statusString":"loaded","type":2,"shards":{"s3010014":["PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1","PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a"]},"status":3,"replicationFactor":2,"path":"","doCompact":true,"name":"_aqlfunctions","shardKeys":["_key"],"distributeShardsLike":"3010001"}}},"Coordinators":{"CRDN-0ea2f097-d05e-4b43-968a-810c87146354":"none","CRDN-83ec9ecb-a66e-45df-8ce6-aa677fd2d92e":"none","CRDN-d00e0350-8108-456f-802c-53c3a8b54174":"none"},"Databases":{"_system":{"id":"1","name":"_system"}},"Version":22,"Singles":{},"Lock":"UNLOCKED","AsyncReplication":{}},"Agency":{"Definition":1},"Supervision":{"DBServers":{},"Shards":{},"Health":{"PRMR-b8f8e681-84d5-4b04-9298-50e30a2eed5a":{"Endpoint":"tcp://192.168.136.130:8530","Host":"e9f0b52f06194937928cbb1f3ed719aa","Timestamp":"2018-05-30T10:18:00Z","Status":"GOOD","SyncStatus":"SERVING","ShortName":"DBServer0003"},"PRMR-c9787481-3587-4fee-8f24-ea7e70823cd1":{"Endpoint":"tcp://192.168.136.131:8530","Host":"e9f0b52f06194937928cbb1f3ed719aa","Timestamp":"2018-05-30T10:18:00Z","Status":"GOOD","SyncStatus":"SERVING","ShortName":"DBServer0002"},"CRDN-d00e0350-8108-456f-802c-53c3a8b54174":{"Endpoint":"tcp://192.168.136.131:8529","Host":"e9f0b52f06194937928cbb1f3ed719aa","Timestamp":"2018-05-30T10:18:01Z","Status":"GOOD","SyncStatus":"SERVING","ShortName":"Coordinator0003"},"CRDN-83ec9ecb-a66e-45df-8ce6-aa677fd2d92e":{"Endpoint":"tcp://192.168.136.132:8529","Host":"e9f0b52f06194937928cbb1f3ed719aa","Timestamp":"2018-05-30T10:18:42Z","Status":"GOOD","SyncStatus":"SERVING","ShortName":"Coordinator0002"},"PRMR-d7b9e6c2-e2ac-4c4c-bcb8-59eddaf182fd":{"Endpoint":"tcp://192.168.136.132:8530","Host":"e9f0b52f06194937928cbb1f3ed719aa","Timestamp":"2018-05-30T10:18:42Z","Status":"GOOD","SyncStatus":"SERVING","ShortName":"DBServer0001"},"CRDN-0ea2f097-d05e-4b43-968a-810c87146354":{"Endpoint":"tcp://192.168.136.130:8529","Host":"e9f0b52f06194937928cbb1f3ed719aa","Timestamp":"2018-05-30T10:17:58Z","Status":"GOOD","SyncStatus":"SERVING","ShortName":"Coordinator0001"}},"State":{"Mode":"Normal","Timestamp":"2018-05-20T11:51:27Z"}},"Bootstrap":"CRDN-d00e0350-8108-456f-802c-53c3a8b54174: done"},"arangodb-helper":{"arangodb":{"master":"http://192.168.136.130:8528/"}}}]
```

**Comment by p30arena** [comment]
@OmarAyo 

> curl -L http://192.168.136.130:8531/_api/agency/read -L -d '[[ "/" ]]' --dump -

`HTTP/1.1 503 Service Unavailable
X-Content-Type-Options: nosniff
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 23

{"message":"No leader"}`

**Comment by OmarAyo** [comment]
Please change the command that I sent you in the previous message with the following:


- For generating the JWT-token please run (for example using _geheim_ as your JWT-secret):
  `jwtgen -a HS256 -s geheim -c server_id=setup -c iss=arangodb`
   and the output will be: 
 `eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE1Mjc2Nzg0MDgsInNlcnZlcl9pZCI6InNldHVwIiwiaXNzIjoiYXJhbmdvZGIifQ.1iS4R5MTkhO1X-PksUXWgotVcdOSYTKMIRt9ZTgpjmY `

I changed server_id=hans with server_id= **setup**

- Now please send the following request:

`curl -L http://IP:PORT/_api/agency/read -d '[["/"]]'  -H "Authorization: bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE1Mjc2Nzg0MDgsInNlcnZlcl9pZCI6InNldHVwIiwiaXNzIjoiYXJhbmdvZGIifQ.1iS4R5MTkhO1X-PksUXWgotVcdOSYTKMIRt9ZTgpjmY"` 


In this case, I changed -H " Authentication: bearer ...." with -H " **Authorization** ..."

Please make sure to use the IP and PORT of one agent, if you have no response try to change the agent PORT (you should have at least 3 diffrent ports)

Best,

**Comment by p30arena** [comment]
@OmarAyo 

> curl -L http://192.168.136.130:8531/_api/agency/read -d '[["/"]]' -H "Authorization: bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE1Mjc3NTA4NzcsInNlcnZlcl9pZCI6InNldHVwIiwiaXNzIjoiYXJhbmdvZGIifQ.LH0aaEcBfnaUVaSF0aPuYF_HcVzzOqdUveTYzdUWb74"

`{"message":"No leader"}`

**Comment by OmarAyo** [comment]
Hi @p30arena,

Please allow me to understand better your situation, as I know you did several tests, and I would like to understand the current situation, especially considering that in a previous message you wrote that you cannot reproduce the error when you tried a fresh start (on a new directory)

a) Is your Cluster up and running right now?
b) If not, could you please run 
   b1) ps auxww | grep arangod 
   and also
   b2) netstat -natp | grep 85[23][8901] 
   b3) firewall-cmd --list-all
   b4) firewall-cmd --list-ports 
   on each machine and let us know the output?
c) Could you also double check that from each machine you can communicate to the others, e.g., assuming we have machine1, machine2, machine3, you can execute:

from machine 1:

telnet machine2 8528
telnet machine2 8529
telnet machine2 8530
telnet machine2 8531

telnet machine3 8528
telnet machine3 8529
telnet machine3 8530
telnet machine3 8531

and similar commands from machine2 to reach machine1 and machine3, and from machine3 to reach machine1 and machine2 

d) If the cluster is not running, please could you:
   d1) upload an archive with all logs (Coordinators, DBServers and Agents)

Regarding the agency dump, it would be useful to know the whole reply (e.g. if you get something before {"message":"No leader"})

**Comment by p30arena** [comment]
Hi @OmarAyo 
I'm sure that there are no network problems, because on a new directory the cluster works fine.

https://www.dropbox.com/s/f37a05joect520q/ara.tar.gz?dl=0

**Comment by dsonet** [comment]
Seems related to authentication = true. When turn it on, this could happen.

**Comment by OmarAyo** [comment]
Hi all,

I have attached a shell script that can be used to collect diagnostics information from your services. If you are seeing again the issue we would kindly ask you to run the script on all your machines and instances. No data will be collected. It's meta data only:

bash arangoinspect.sh <endpoint> [secret file name]

For example: 
bash arangoinspect https://machine-a:8529 /opt/secret.txt

Run it on all your Coordinators, DBServer and Agents (use corresponding ports)

It can only run successfully if the following programs are installed: jq, curl, npm, node, nodejs-legacy
Additionally, npm install -g jwtgen is needed to be able to communicate with the nodes, if authentication is enabled your Cluster.

After running the script you should have 3 archives for each node. Please share the archives. If one of the nodes is down, just post archives from the survived nodes.

[arangoinspect_.zip](https://github.com/arangodb/arangodb/files/2127674/arangoinspect_.zip)

**Comment by dsonet** [comment]
I'll try the inspect later.
I need to describe what I encountered first.
I use the latest version 3.3.10, Ubuntu 16.04, mmfiles, all settings are defaults, setup 3 nodes.
At the beginning, I shutdown 1 random node each time, the cluster can recognize  the node, and still runs well. But the recent time I tried the same action(the cluster already up about a week), it doesn't work out as before. The cluster is gone after 1 node failed(which shouldn't). There is nothing changed(I mean system env, network, etc...)

**Comment by sleto-it** [comment]
@dsonet ,

Thanks for the additional logs. We believe we have identified a possible bug, and we are performing additional analysis and testing. There is a PR (https://github.com/arangodb/arangodb/pull/5686) open that when merged will hopefully fix your issues

We will keep you updated

Many thanks,

**Comment by jeppeband** [comment]
I am having the exact same issue in my cluster setup. And the only solution I have found so far is to remove everything that has to do with Arango and reinstall it. So I was wondering if you found a better way of doing it, while we are waiting for the bug to be fixed?

**Comment by snowzjy** [comment]
I am having the exact same issue in my cluster also. 
machine1,machine2,machine3 have save issue:

2018/07/09 12:08:04 Failed to get master URL, retrying in 5sec (All 3 attemps resulted in temporary failure)
2018/07/09 12:08:19 Failed to get master URL, retrying in 5sec (All 3 attemps resulted in temporary failure)
2018/07/09 12:08:34 Failed to get master URL, retrying in 5sec (All 3 attemps resulted in temporary failure)
......


The firewall and iptables are disabled. And I tred the commands from machine 1 /2/3 to other machines like this :

telnet machine2 8528
telnet machine2 8529
telnet machine2 8530
telnet machine2 8531

telnet machine3 8528
telnet machine3 8529
telnet machine3 8530
telnet machine3 8531

8528 and 8531 can connected. But 8529 and 8530 have error ：
telnet: connect to address xxx.xxx.xxx.xxx: Connection refused

I think this is a same issue maybe.

**Comment by edmondboulet** [comment]
Hello, same here.

Here is my master log ( i followed this steps from the [doc](https://docs.arangodb.com/3.3/Manual/Deployment/ArangoDBStarter.html#automatic-docker-clusters)):

```
[user@srv-poc01 data]$ sudo docker run -it --name=adb1 --rm -p 8528:8528     -v arangodb1:/data     -v /var/run/docker.sock:/var/run/docker.sock     arangodb/arangodb-starter     --starter.address=10.31.127.21
2018-07-12T09:55:10Z |INFO| Starting arangodb version 0.12.0, build cd81a60 component=arangodb
2018-07-12T09:55:10Z |INFO| Serving as master with ID 'eff953ff' on 10.31.127.21:8528... component=arangodb
2018-07-12T09:55:10Z |INFO| Waiting for 3 servers to show up.
 component=arangodb
2018-07-12T09:55:10Z |INFO| Use the following commands to start other servers: component=arangodb

docker volume create arangodb2 && \
    docker run -it --name=adb2 --rm -p 8538:8528 -v arangodb2:/data \
    -v /var/run/docker.sock:/var/run/docker.sock arangodb/arangodb-starter:latest \
    --starter.address=10.31.127.21 --starter.join=10.31.127.21
2018-07-12T09:55:10Z |INFO| ArangoDB Starter listening on 0.0.0.0:8528 (10.31.127.21:8528) component=arangodb

docker volume create arangodb3 && \
    docker run -it --name=adb3 --rm -p 8548:8528 -v arangodb3:/data \
    -v /var/run/docker.sock:/var/run/docker.sock arangodb/arangodb-starter:latest \
    --starter.address=10.31.127.21 --starter.join=10.31.127.21

2018-07-12T09:56:00Z |INFO| Added new peer '2da52dec': 10.31.127.21, portOffset: 0 component=arangodb
2018-07-12T09:57:34Z |INFO| Added new peer '98667237': 10.31.127.21, portOffset: 0 component=arangodb
2018-07-12T09:57:35Z |INFO| Starting service... component=arangodb
2018-07-12T09:57:35Z |INFO| Looking for a running instance of agent on port 8531 component=arangodb
2018-07-12T09:57:35Z |INFO| Starting agent on port 8531 component=arangodb
2018-07-12T09:57:36Z |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2018-07-12T09:57:36Z |INFO| Starting dbserver on port 8530 component=arangodb
2018-07-12T09:57:37Z |INFO| Looking for a running instance of coordinator on port 8529 component=arangodb
2018-07-12T09:57:37Z |INFO| Starting coordinator on port 8529 component=arangodb
2018-07-12T09:58:11Z |INFO| agent up and running (version 3.3.11). component=arangodb
2018-07-12T09:58:45Z |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-07-12T09:59:00Z |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"

```

**Comment by OmarAyo** [comment]
Hi everyone,

@snowzjy:
could you please attach all your logs from all nodes? It could be also useful if you share with us the commands you are using for deploying your Cluster, are you using the Starter or deploying it manually? Please note that I've attached in this ticket the script for arangoinspect.sh - please could you run it on all your machines and all endpoints.
You should pass the path of the jwt file. The command will create a json file that we can analyze to understand better the status.

Examples of the commands:
```
sudo apt -y install jq npm
./arangoinspect.sh localhost:8529 secret.txt
./arangoinspect.sh localhost:8530 secret.txt
./arangoinspect.sh localhost:8531 secret.txt
```
@edmondboulet: 
to understand better your case, I'm just wondering if did you use the exact command reported from the documentation ? Asking this because your fist command seems to miss part of the entire command reported in the documentation, please see below:

You reported the following:
```
[user@srv-poc01 data]$ sudo docker run -it --name=adb1 --rm -p 8528:8528   -v arangodb1:/data 

```
And missing: `-v /var/run/docker.sock:/var/run/docker.sock  arangodb/arangodb-starter \ --starter.address=192.168.1.32`

In addition, please attach all the logs you have from all nodes for a better analyze 


@jeppeband & @dsonet 
How things are going now, are you facing again this type of issue, in case please provide the logs from all the nodes.

@p30arena 
We haven't heard back from you recently, are you still facing this issue. Could you please run the arangoinspect.sh script provided in this ticket.

Looking forward to all your replies 

Best,

**Comment by p30arena** [comment]
@OmarAyo 

[ara.zip](https://github.com/arangodb/arangodb/files/2201320/ara.zip)

**Comment by OmarAyo** [comment]
Hi @snowzjy 

Thanks for using the _arangoinspect_ and providing us the files. In addition, as mentioned in my previous message, could you provide us the followings
- all the logs from all the nodes
- Are you using the _Starter_ for deploying the Cluster or are you deploying it manually?
- Exact commands you are using for deploying the Cluster

Hi @p30arena 

Thanks for attaching all the requirements. I'm just wondering, how things are going so far, are you still facing the issue ? We had a look into the logs that you provided us on 1st June, but unfortunately they are only for just 1 node, for a better analysis we would like to have all the logs from all the nodes of your Cluster   

Hi @dsonet, @edmondboulet, @jeppeband 

Could you please share with us a quick feedback? In case you are still facing the reported issue, please provide us what mentioned in my previous message?

Thanks,

**Comment by maxkernbach** [comment]
Hi everyone,

the latest release 3.3.12 includes an agency fix (see PR [5743](https://github.com/arangodb/arangodb/pull/5743/files)), which also might fix the `Failed to get master URL` issue.
One of our customers who has upgraded to 3.3.12 has reported that they have not seen this problem since they upgraded.

Could you please try to upgrade 3.2.12 and check if the problem persists?

**Comment by snowzjy** [comment]
Hi @maxkernbach good message! good version! I installed 3.3.12 and start my cluster by starter, the database load faster! There are no errors for now , I will test some days.

**Comment by p30arena** [comment]
I will check later, and close the issue if resolved.

**Comment by seansilvestri** [comment]
I'm curious what was the issue?

**Comment by edmondboulet** [comment]
Hi @OmarAyo,

I miss-typed, the command I use to create the first server instance is :

`docker run -it --name=adb1 --rm -p 8528:8528     -v arangodb1:/data     -v /var/run/docker.sock:/var/run/docker.sock     arangodb/arangodb-starter     --starter.address=10.31.127.24`

10.31.127.24 being the IP of the machine on the network and the other machines can totally ping it.
I just upgraded to 3.3.13 Enterprise and the issue is still here. 


Also now it seems the master can detect the other instance briefly :

```
2018-08-08T13:52:20Z |INFO| Added new peer 'cf544e61': 10.31.127.24, portOffset: 0 component=arangodb
2018-08-08T13:53:46Z |INFO| Added new peer 'a391afae': 10.31.127.24, portOffset: 10 component=arangodb

```

Thanks for the help.

**Comment by Napsty** [comment]
_Seems_ I came across the same bug when trying to create a cluster from a standalone server with an existing data directory using rocksdb engine (please do let me know if this is not the same bug):

```
root@arango1:~# arangodb --starter.address=192.168.15.161 --starter.data-dir=/var/lib/arangodb3 --server.storage-engine=rocksdb --log.verbose
2018-08-09T21:19:52+02:00 |INFO| Starting arangodb version 0.12.0, build cd81a60 component=arangodb
2018-08-09T21:19:52+02:00 |DEBU| Using /usr/sbin/arangod as default arangod executable. component=arangodb
2018-08-09T21:19:52+02:00 |DEBU| Using /usr/share/arangodb3/js as default JS dir. component=arangodb
2018-08-09T21:19:52+02:00 |DEBU| Using process runner component=arangodb
2018-08-09T21:19:52+02:00 |INFO| Relaunching service with id '9a70cd36' on 192.168.15.161:8528... component=arangodb
2018-08-09T21:19:52+02:00 |DEBU| hostPort=8528 masterPort=8528 #AllPeers=3 component=arangodb
2018-08-09T21:19:52+02:00 |INFO| ArangoDB Starter listening on 0.0.0.0:8528 (192.168.15.161:8528) component=arangodb
2018-08-09T21:19:52+02:00 |INFO| Looking for a running instance of agent on port 8531 component=arangodb
2018-08-09T21:19:52+02:00 |DEBU| Cannot find /var/lib/arangodb3/agent8531/data/LOCK component=arangodb
2018-08-09T21:19:52+02:00 |INFO| Starting agent on port 8531 component=arangodb
2018-08-09T21:19:52+02:00 |DEBU| Waiting on 8086 component=arangodb
2018-08-09T21:19:53+02:00 |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2018-08-09T21:19:53+02:00 |DEBU| Cannot find /var/lib/arangodb3/dbserver8530/data/LOCK component=arangodb
2018-08-09T21:19:53+02:00 |INFO| Starting dbserver on port 8530 component=arangodb
2018-08-09T21:19:53+02:00 |DEBU| Waiting on 8130 component=arangodb
2018-08-09T21:19:54+02:00 |INFO| Looking for a running instance of coordinator on port 8529 component=arangodb
2018-08-09T21:19:54+02:00 |DEBU| Cannot find /var/lib/arangodb3/coordinator8529/data/LOCK component=arangodb
2018-08-09T21:19:54+02:00 |INFO| Starting coordinator on port 8529 component=arangodb
2018-08-09T21:19:54+02:00 |DEBU| Waiting on 8143 component=arangodb
2018-08-09T21:19:54+02:00 |INFO| agent up and running (version 3.3.13). component=arangodb
2018-08-09T21:20:02+02:00 |DEBU| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:20:17+02:00 |DEBU| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:20:32+02:00 |DEBU| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:20:47+02:00 |DEBU| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:21:02+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:21:17+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:21:32+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:21:47+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:22:02+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:22:17+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:22:24+02:00 |WARN| dbserver not ready after 5min!: Status trail: []int{} component=arangodb
2018-08-09T21:22:25+02:00 |WARN| coordinator not ready after 5min!: Status trail: []int{} component=arangodb
2018-08-09T21:22:32+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:22:47+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:23:02+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:23:17+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:23:32+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:23:47+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:24:02+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:24:17+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:24:32+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
2018-08-09T21:24:47+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
^CReceived signal: interrupt
2018-08-09T21:24:57+02:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context canceled"
2018-08-09T21:24:57+02:00 |INFO| Shutting down services... component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| Terminating coordinator... component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| Waiting on 8143 component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| Wait on 8143 result component=arangodb
2018-08-09T21:24:57+02:00 |INFO| coordinator has terminated component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| Wait on 8143 result component=arangodb error="waitid: no child processes"
2018-08-09T21:24:57+02:00 |DEBU| coordinator terminated component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| Terminating dbserver... component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| Waiting on 8130 component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| Wait on 8130 result component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| dbserver terminated component=arangodb
2018-08-09T21:24:57+02:00 |DEBU| Wait on 8130 result component=arangodb error="wait: no child processes"
2018-08-09T21:24:57+02:00 |INFO| dbserver has terminated component=arangodb
^[[A2018-08-09T21:25:00+02:00 |DEBU| Terminating agent... component=arangodb
2018-08-09T21:25:00+02:00 |DEBU| Waiting on 8086 component=arangodb
2018-08-09T21:25:00+02:00 |DEBU| Wait on 8086 result component=arangodb
2018-08-09T21:25:00+02:00 |INFO| agent has terminated component=arangodb
2018-08-09T21:25:00+02:00 |DEBU| Wait on 8086 result component=arangodb error="waitid: no child processes"
2018-08-09T21:25:00+02:00 |DEBU| agent terminated component=arangodb
```

Although this is the cluster starter (master), it does not start up correctly. Oh and yes, ArangoDB service was stopped prior launching the arangodb command. 

This happens on a fresh install 3.3.13 on Debian 8 (jessie) from .deb package.

**Comment by OmarAyo** [comment]
Hi @Napsty 

I'm a bit confused, did you try to start your Cluster on top of a standalone directory? Was your standalone instance using `mmFiles` or `RocksDB` as storage-engine? 

Could you please specify the followings:
- all your Starter command used for deploying the Cluster
- was the Standalone instance stopped before trying to deploy the Cluster
- all your log files 

I recommend you open a separate ticket for this issue since it seems, IMHO, not related to this ticket.

What I suggest you is to deploy your Cluster on an empty directory using the following [procedure](https://docs.arangodb.com/devel/Manual/Deployment/Cluster/UsingTheStarter.html)

**Comment by Napsty** [comment]
> Was your standalone instance using mmFiles or RocksDB as storage-engine?
It was using RocksDB

> all your Starter command used for deploying the Cluster

```
root@arango1:~# arangodb --starter.address=192.168.15.161 --starter.data-dir=/var/lib/arangodb3 --server.storage-engine=rocksdb --log.verbose

root@arango2:~# arangodb --starter.join=192.168.15.161 --starter.data-dir=/var/lib/arangodb3 --server.storage-engine=rocksdb --log.verbose

root@arango3:~# arangodb --starter.join=192.168.15.161 --starter.data-dir=/var/lib/arangodb3 --server.storage-engine=rocksdb --log.verbose
```

> was the Standalone instance stopped before trying to deploy the Cluster
Yes

> deploy your Cluster on an empty directory
So does that mean, that it's not possible to create an ArangoDB cluster out of a standalone ArangoDB server? When I use an empty directory as data dir, the existing databases and collections from the standalone ArangoDB are gone of course. A clarification on this topic would help a lot.

**Comment by OmarAyo** [comment]
Hi @Napsty 

As I said above, please open a new ticket on GitHub with the following question, this will keep topics separated and easily reachable. 

If you want I can open a ticket on behalf of you and we can continue discussing there 

Thanks,

**Comment by edmondboulet** [comment]
Hi @OmarAyo 

I just realized I forgot to attach the logs from the each nodes, here they are :

[node2.txt](https://github.com/arangodb/arangodb/files/2277926/node2.txt)
[node3.txt](https://github.com/arangodb/arangodb/files/2277927/node3.txt)
[node1.txt](https://github.com/arangodb/arangodb/files/2277928/node1.txt)

I would gladly run _arangoinspect_ but I can't for the moment.

Thanks for the help.

**Comment by p30arena** [comment]
@OmarAyo 
I upgraded to 3.3.13

`|INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"`
[arangoinspect-97a57f183199abe1fc3a230224739fa4-dmesg.txt](https://github.com/arangodb/arangodb/files/2280020/arangoinspect-97a57f183199abe1fc3a230224739fa4-dmesg.txt)
[arangoinspect-97a57f183199abe1fc3a230224739fa4-ip.txt](https://github.com/arangodb/arangodb/files/2280021/arangoinspect-97a57f183199abe1fc3a230224739fa4-ip.txt)
[arangoinspect-97a57f183199abe1fc3a230224739fa4-meminfo.txt](https://github.com/arangodb/arangodb/files/2280022/arangoinspect-97a57f183199abe1fc3a230224739fa4-meminfo.txt)
[arangoinspect-97a57f183199abe1fc3a230224739fa4-uname.txt](https://github.com/arangodb/arangodb/files/2280023/arangoinspect-97a57f183199abe1fc3a230224739fa4-uname.txt)
[arangoinspect-97a57f183199abe1fc3a230224739fa4-uptime.txt](https://github.com/arangodb/arangodb/files/2280024/arangoinspect-97a57f183199abe1fc3a230224739fa4-uptime.txt)
[arangoinspect-97a57f183199abe1fc3a230224739fa4-date.txt](https://github.com/arangodb/arangodb/files/2280025/arangoinspect-97a57f183199abe1fc3a230224739fa4-date.txt)
[arangoinspect-97a57f183199abe1fc3a230224739fa4-df.txt](https://github.com/arangodb/arangodb/files/2280026/arangoinspect-97a57f183199abe1fc3a230224739fa4-df.txt)

**Comment by OmarAyo** [comment]
Hi @p30arena 

Thanks for providing us the `arangoinspect` output. We will have a look and get back to you.

Best,

**Comment by p30arena** [comment]
@OmarAyo 
I can dump the container directory and send it here,
would it be useful?

**Comment by rafis** [comment]
Try: `unset http_proxy https_proxy no_proxy HTTP_PROXY HTTPS_PROXY NO_PROXY`

**Comment by Himan000** [comment]
@OmarAyo , No further solution about this problem?

**Comment by mazumderd** [comment]
I am getting same error interestingly only when we try to add the last node to cluster, other nodes get added pretty easily
